clang/include/clang/3C/ProgramInfo.h

```diff
@@ -146,6 +146,12 @@ class ProgramInfo : public ProgramVariableAdder {
   void registerTranslationUnits(
       const std::vector<std::unique_ptr<clang::ASTUnit>> &ASTs);
 
+  // Hack to allow `emit` in RewriteUtils.cpp to avoid printing the main file to
+  // stdout multiple times in the edge case of a compilation database containing
+  // multiple translation units for the main file
+  // (https://github.com/correctcomputation/checkedc-clang/issues/374#issuecomment-893612654).
+  bool StdoutModeEmittedMainFile = false;
```

Comment:
A thought I've had in the past is that we need a better way to pass information between translation units within a single pass without adding it to the global `ProgramInfo` class (which is visible across all passes). I'm not holding up this change based on this comment, but I'd be interested in any thoughts you have on it.

Reply:
Now that you bring it up, I realize that every pass uses an object of a subclass of `ASTConsumer` that persists across all translation units, so we could store the data in those objects.  (The use of `ASTConsumer` is a hold-over from before #488, when LibTooling required us to provide an `ASTConsumer` to the `ClangTool`.  Now that we no longer have that requirement, I had been thinking we could just replace the objects with global functions, since all of them are currently stateless.  But since you point out that we may want to add state to some of these objects, maybe it makes sense to keep all of them as objects and just replace `ASTConsumer` with our own simpler interface if we want.)

Of course, this doesn't help with data that is needed by multiple passes but still feels too purpose-specific for the `ProgramInfo` (and I may be adding some such data in #657), but we can look for a better solution for that case later as desired.

I went ahead and moved the `StdoutModeEmittedMainFile` flag to `RewriteConsumer` in this PR; that's much nicer.